test/dataframe.jl

```diff
@@ -662,4 +662,66 @@ module TestDataFrame
             @test x == y
         end
     end
+
+    @testset "permutecols!" begin
+        a, b, c = 1:5, 2:6, 3:7
+        original = DataFrame(a=a, b=b, c=c)
+
+        df = deepcopy(original)
+        expected = deepcopy(original)
+        permutecols!(df, [:a, :b, :c])
+        @test df == expected
+        permutecols!(df, 1:3)
+        @test df == expected
+
+        df = deepcopy(original)
+        expected = DataFrame(b=b, c=c, a=a)
+        permutecols!(df, [:b, :c, :a])
+        @test df == expected
+        df = deepcopy(original)
+        permutecols!(df, [2, 3, 1])
+        @test df == expected
+
+        df = deepcopy(original)
+        expected = DataFrame(c=c, a=a, b=b)
+        permutecols!(df, [:c, :a, :b])
+        @test df == expected
+        df = deepcopy(original)
+        permutecols!(df, [3, 1, 2])
+        @test df == expected
+
+        df = deepcopy(original)
+        expected = DataFrame(a=a, c=c, b=b)
+        permutecols!(df, [:a, :c, :b])
+        @test df == expected
+        df = deepcopy(original)
+        permutecols!(df, [1, 3, 2])
+        @test df == expected
+
+        df = deepcopy(original)
+        expected = DataFrame(b=b, a=a, c=c)
+        permutecols!(df, [:b, :a, :c])
+        @test df == expected
+        df = deepcopy(original)
+        permutecols!(df, [2, 1, 3])
+        @test df == expected
+
+        df = deepcopy(original)
+        expected = DataFrame(c=c, b=b, a=a)
+        permutecols!(df, [:c, :b, :a])
+        @test df == expected
+        df = deepcopy(original)
+        permutecols!(df, [3, 2, 1])
+        @test df == expected
+
+        # Invalid
+        df = DataFrame(a=a, b=b, c=c)
+        @test_throws Exception permutecols!(df, [:a, :b])
```

Comment:
Once `error` is changed, this should be `ArgumentError` instead of `Exception`.

Reply:
Sure thing.

Not sure why the tests are failing. They're passing locally. I'll do a `Pkg.update()` and see if something's changed in one of the requirements that might be affecting things. The error appears to be happening before Travis even gets to the `permutecols!` tests. (Line 376 of test/dataframe.jl, looks like some stack/unstack tests.)

Reply:
After updating everything, I'm now able to replicate the error locally. I'm still not sure what's causing it, however. It's appearing as part of an `isequals` check earlier in the tests. I checked out master and it appears to be an issue on master as well, so I don't think that it's related to my changes.